.github/workflows/build-test-distribute.yaml

```
@@ -160,6 +246,109 @@ jobs:
           trap on_exit EXIT
           make docker/push
           make docker/manifest
+          manifests=$(make manifests/release/show | jq -R -c -S 'split("\n")') # Create JSON array for release manifests needed for signing
+          echo "$published_manifests=${manifests}"
+          echo "manifests=${manifests}" >> "$GITHUB_OUTPUT"
+  # Keyless container image signing 
+  # This step will sign docker manifest image recursively for all platforms by digest after publishing the image
+  # Publishes all signatures to the "kumahq/notary" public registry until TODO above
+  # Record permanent entry to Sigstore Public Rekor - Transparency log
+  # Note:
+  #  For private repos, no good solution is available for transparency requirement for SLSA.
+  #  Hence exposing repo name and workflow details and commit details by leveraging public sigstore instance
+  #  Generate provenance and image signing separately for every image after publishing. 
+  # TODO:
+  #   Split sigining step and notary registries to avoid customers viewing signatures of internal images
+  #     Official image signatures: kumahq/notary
+  #     Internal/Preview image signatures:kumahq/notary-internal
+  sign-image-manifests:
+    permissions:
+      id-token: write # needed for signing the images with GitHub OIDC Token
+    needs: ["metadata", "distribute-images"]
+    if: >
+      github.repository_owner == 'Kong'  && needs.metadata.outputs.ALLOW_PUSH == 'true'
+
+    runs-on: ubuntu-latest-kong
```

not going to work for OSS (we have stuff to make this run on our runners for KM).
```suggestion
    runs-on: ubuntu-latest
```

---

done